adm_program/system/classes/user.php

```diff
@@ -1213,9 +1213,20 @@ public function setPassword($newPassword, $isNewPassword = false, $doHashing = t
                 $cost = (int) $gPreferences['system_hashing_cost'];
             }
 
-            $newPassword = PasswordHashing::hash($newPassword, PASSWORD_DEFAULT, array('cost' => $cost));
+            if (is_int($gPasswordHashAlgorithm))
+            {
+                $newPasswordHash = PasswordHashing::hash($newPassword, $gPasswordHashAlgorithm, array('cost' => $cost));
+            }
+            elseif (is_string($gPasswordHashAlgorithm))
+            {
+                $newPasswordHash = PasswordHashing::hash($newPassword, $gPasswordHashAlgorithm, array('rounds' => 5000));
+            }
+            else
+            {
+                $newPasswordHash = PasswordHashing::hash($newPassword, PASSWORD_DEFAULT, array('cost' => $cost));
+            }
 
-            if ($newPassword === false)
+            if ($newPasswordHash === false)
```

Comment:
You should do something with `$newPasswordHash` here, at the moment the plain `$newPassword` gets saved.